mainnet-contracts/src/interface/IWithdrawalManager.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity >=0.8.0 <0.9.0;
+
+import { PufferVaultV3 } from "../PufferVaultV3.sol";
+
+/**
+ * @title IWithdrawalManager
+ * @author Puffer Finance
+ * @notice Interface for the WithdrawalManager contract
+ * @custom:security-contact security@puffer.fi
+ */
+interface IWithdrawalManager {
+    /// @notice Thrown when attempting to finalize a batch that has already been finalized
+    error BatchAlreadyFinalized();
+
+    /// @notice Thrown when attempting to complete a withdrawal that hasn't been finalized yet
+    error NotFinalized();
+
+    /// @notice Thrown when attempting to finalize a batch that isn't full yet
+    error BatchNotFull();
+
+    /// @notice Thrown when attempting to withdraw an amount below the minimum threshold
+    error WithdrawalAmountTooLow();
+
+    /// @notice Emitted when a withdrawal is requested
+    /// @param batchIndex The index of the batch the withdrawal is added to
+    /// @param pufETHAmount The amount of pufETH requested for withdrawal
+    /// @param recipient The address that will receive the withdrawn ETH
+    event WithdrawalRequested(uint256 indexed batchIndex, uint256 pufETHAmount, address indexed recipient);
```

Comment:
We are missing withdrawer

Reply:
Do we need it in the event? It would be nice to have, but that requires us to add the withdrawer to struct Withdrawal
In that case you can pack it like 
{
    address withdrawer; // slot 0
    uint96 pufethAmount // slot 0
    address recipient // slot 1
    uint96 pufETHToEthExchangeRate; // slot 1
}

Reply:
Maybe `withdrawalIdx` makes more sense.

Reply:
the idx should be good. msg.sender is the withdrawer